test/test_producer.py

```diff
@@ -57,6 +58,8 @@ def test_producer_async_queue_overfilled_batch_send(self):
             message_list = [message] * (queue_size + 1)
             producer.send_messages(topic, partition, *message_list)
         self.assertEqual(producer.queue.qsize(), queue_size)
+        for _ in xrange(producer.queue.qsize()):
+            producer.queue.get()
```

Comment:
this is needed to actually drain the queue and avoid a hanging test b/c the async thread never finishes